tezos/encoding/src/binary_reader.rs

```diff
@@ -251,7 +251,7 @@ impl BinaryReader {
                 let mut buf_slice = safe!(buf, upper, buf.take(upper));
                 let res = self.decode_value(&mut buf_slice, inner_encoding);
                 match res {
-                    Err(BinaryReaderError::Underflow { bytes: _ }) => {
+                    Err(BinaryReaderError::Underflow { bytes: _ }) if upper == *max => {
```

Comment:
It is weird seeing things go from an underflow error to a "boundary exceeded" one. Reasoning about what the code is doing here makes it clear, but a comment explaining that in this case the inner underflow is caused by the outer-imposed limit would make it easier to understand.

Reply:
@akoptelov looks good, thanks